bioconda_utils/lint/checks.py

```diff
@@ -0,0 +1,552 @@
+            self.message(section='about')
+
+
+class missing_license(LintCheck):
+    """The recipe is missing the ``about/license`` key.
+
+    Please add::
+
+        about:
+           license: <name of license>
+
+    """
+    def check_recipe(self, recipe):
+        if not recipe.get('about/license', ''):
+            self.message(section='about')
+
+
+class missing_tests(LintCheck):
+    """The recipe is missing tests.
+
+    Please add::
+
+        test:
+            commands:
+               - some_command
+
+    and/or::
+
+        test:
+            imports:
+               - some_module
+
+
+    and/or any file named ``run_test.py`, ``run_test.sh`` or
+    ``run_test.pl`` executing tests.
+
+    """
+    test_files = ['run_test.py', 'run_test.sh', 'run_test.pl']
+
+    def check_recipe(self, recipe):
+        if any(os.path.exists(os.path.join(recipe.dir, f))
+               for f in self.test_files):
+            return
+        if recipe.get('test/commands', '') or recipe.get('test/imports', ''):
+            return
+        if recipe.get('test', False) is not False:
+            self.message(section='test')
+        else:
+            self.message()
+
+
+class missing_hash(LintCheck):
+    """The recipe is missing a checksum for a source file
+
+    Please add::
+
+       source:
+         sha256: checksum-value
+
+    """
+    checksum_names = ('md5', 'sha1', 'sha256')
+
+    def check_source(self, source, section):
+        if not any(source.get(chk) for chk in self.checksum_names):
+            self.message(section=section)
+
+
+class uses_git_url(LintCheck):
+    """The recipe downloads source from git
+
+    Please build from source archives and don't use the ``git_url``
+    feature of conda.
+
+    """
+    def check_source(self, source, section):
+        if 'git_url' in source:
+            self.message(section=section + '/git_url')
+
+
+class uses_perl_threaded(LintCheck):
+    """The recipe uses ``perl-threaded``
+
+    Please use ``perl`` instead.
+
+    """
+    def check_recipe(self, recipe):
+        if 'perl-threaded' in recipe.get_deps():
+            self.message()
+
+
+class uses_javajdk(LintCheck):
+    """The recipe uses ``java-jdk``
+
+    Please use ``openjdk`` instead.
+
+    """
+    def check_recipe(self, recipe):
+        if 'java-jdk' in recipe.get_deps():
+            self.message()
+
+
+class uses_setuptools(LintCheck):
+    """The recipe uses setuptools in run depends
+
+    Most Python packages only need setuptools during installation.
+    Check if the package really needs setuptools (e.g. because it uses
+    pkg_resources or setuptools console scripts).
+
+    """
+    severity = INFO
+
+    def check_recipe(self, recipe):
+        if 'setuptools' in recipe.get_deps('run'):
+            self.message()
+
+
+class has_windows_bat_file(LintCheck):
+    """The recipe directory contains a ``bat`` file.
+
+    Bioconda does not and will never build Windows packages.
```

Comment:
"and never will" is a bit harsh . . . I'd prefer "does not plan to"

Reply:
It was just my honest estimation. It's just too complicated. Editing as requested, though, wasn't meant to sound harsh.

Reply:
Fixed in e7a10f9